### PR TITLE
Colored coin issuing protocol improved. (EC-319)

### PR DESCRIPTION
**Pull request by oskin1**

### CODE REVIEW

File: src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala [hunk 1]
```diff
@@ -1,37 +1,32 @@
...
-  override def boxes(digest: Digest32, idx: Int): Seq[EncryBaseBox] = {
-    val assetCreationBox = AssetCreationBox(EncryProposition.accountLock(Account(PublicKey @@ Random.randomBytes(32))),
-      Utils.nonceFromDigest(digest ++ Ints.toByteArray(idx) ++ Ints.toByteArray(1)), amount, symbol)
-    Seq(
-      assetCreationBox,
-      AssetIssuingBox(EncryProposition(contractHash),
-        Utils.nonceFromDigest(digest ++ Ints.toByteArray(idx) ++ Ints.toByteArray(2)), amount, assetCreationBox.id)
-    )
-  }
+  override def boxes(digest: Digest32, idx: Int): Seq[EncryBaseBox] =
```
Comment (kapinuss) [inline src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala]: it looks like apply fun

File: src/main/scala/encry/modifiers/state/box/AssetBox.scala [hunk 2]
```diff
@@ -4,26 +4,21 @@ import com.google.common.primitives.{Bytes, Longs, Shorts}
...
+import encry.modifiers.state.box.TokenIssuingBox.TokenId
 import encry.settings.{Algos, Constants}
 import io.circe.Encoder
 import io.circe.syntax._
 import org.encryfoundation.prismlang.core.Types
 import org.encryfoundation.prismlang.core.wrapped.{PObject, PValue}
-import encry.modifiers.state.box.Box.Amount
-import scorex.crypto.authds
-import scorex.crypto.authds.ADKey
-import supertagged.@@
```
Comment (kapinuss) [inline src/main/scala/encry/modifiers/state/box/AssetBox.scala]: !!

File: src/main/scala/encry/modifiers/state/box/TokenIssuingBox.scala [hunk 3]
```diff
@@ -2,28 +2,25 @@ package encry.modifiers.state.box
...
 import encry.modifiers.state.box.EncryBox.BxTypeId
 import encry.settings.{Algos, Constants}
 import io.circe.Encoder
 import io.circe.syntax._
 import org.encryfoundation.prismlang.core.Types
 import org.encryfoundation.prismlang.core.wrapped.{PObject, PValue}
-import encry.modifiers.state.box.Box.Amount
-import scorex.crypto.authds
-import scorex.crypto.authds.ADKey
-import supertagged.@@
```
Comment (kapinuss) [inline src/main/scala/encry/modifiers/state/box/TokenIssuingBox.scala]: !

File: src/main/scala/encry/view/state/UtxoState.scala [hunk 4]
```diff
@@ -23,6 +25,7 @@ import io.iohk.iodb.{ByteArrayWrapper, LSMStore, Store}
 import scorex.crypto.authds._
 import scorex.crypto.authds.avltree.batch._
 import scorex.crypto.hash.Digest32
+
```
Comment (kapinuss) [inline src/main/scala/encry/view/state/UtxoState.scala]: !

File: src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala [hunk 5]
```diff
@@ -1,37 +1,32 @@
...
-import encry.modifiers.state.box.{AssetCreationBox, AssetIssuingBox, EncryBaseBox, EncryProposition}
+import encry.modifiers.state.box.{TokenIssuingBox, EncryBaseBox, EncryProposition}
 import encry.settings.{Algos, Constants}
 import encry.utils.Utils
 import io.circe.syntax._
 import io.circe.{Decoder, Encoder, HCursor}
 import org.encryfoundation.prismlang.compiler.CompiledContract.ContractHash
 import scorex.crypto.hash.Digest32
-import scorex.crypto.signatures.PublicKey
-import scorex.utils.Random
```
Comment (Bromel777) [inline src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala]: Empty string

File: src/main/scala/encry/utils/ByteStr.scala [hunk 6]
```diff
@@ -1,5 +1,6 @@
 package encry.utils
 
+import encry.settings.Algos
 import scorex.crypto.encode.Base58
```
Comment (Bromel777) [inline src/main/scala/encry/utils/ByteStr.scala]: empty string

File: src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala (hunk 5 above)
Comment (GRYE) [inline src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala]: blank line

File: src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala [hunk 7]
```diff
@@ -53,7 +47,7 @@ object AssetIssuingDirective {
       amount <- c.downField("amount").as[Long]
       symbol <- c.downField("symbol").as[String]
```
Comment (GRYE) [inline src/main/scala/encry/modifiers/mempool/directive/AssetIssuingDirective.scala]: Remove `symbol <- c.downField("symbol").as[String] `

**Pull request merged**